Src/FluentAssertions/Common/StringExtensions.cs

```diff
@@ -41,6 +42,14 @@ public static string IndexedSegmentAt(this string value, int index)
             return $"{formattedString} (index {index})".EscapePlaceholders();
         }
 
+        /// <summary>
```

Comment:
Consider adding
```cs
public static bool ContainsSpecificCollectionIndex(this string indexedPath)
    => Regex.IsMatch(indexedPath, @"\[\d+\]", "[]");
```
such that `MemberPath.ContainsSpecificCollectionIndex()` simply becomes
```cs
dottedPath.ContainsSpecificCollectionIndex()
```